pkg/client/dtclient/settings_client.go

```diff
@@ -187,6 +190,13 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 		}
 	}
 
+	if match, err := d.findObjectWithMatchingConstraints(ctx, obj); err != nil {
+		return DynatraceEntity{}, err
+	} else if match != nil {
```

Comment:
minor: unnecessary else - we return anyway before

Reply:
Why do we return anyway before?
Frankly I haven't double-checked if it's impossible at the moment, but generally I think it's fine to assume that `findObjectWithMatchingConstraints` might return a `nil` entity even if there was no error

Reply:
Actually just saw while debugging an issue my recent changes seem to have introduced: 
Errors are only returned for actual errors. If there's no match found, the method will return `nil, nil`, no error, but also no match. So this is if/else seems to do exactly what it should - unless I'm missing something